src/db/util/legacy-versioned-model.ts

```diff
@@ -42,23 +47,59 @@ export type FieldsWithVersioned<
  * This definition function extends the given table definition with columns that
  * are present on all models defined by the function `versionLib.versionModel`.
  */
-export const defineLegacyVersionedModel = <
-  F extends FieldDefinition,
-  IDField extends string &
-    (keyof F['generated'] | keyof F['generatedCompositeKey']),
-  SoftDeletionEnabled extends boolean,
->(opts: {
-  tableName: string;
-  fields: F;
-  idField: IDField;
-  softDeletionEnabled: SoftDeletionEnabled;
-}): ModelWithIdInitializer<
-  FieldsWithVersioned<F, SoftDeletionEnabled>,
-  IDField
-> => {
-  const fields: ExtendedFields<F> = merge({}, opts.fields, VERSIONED_FIELDS);
-  return defineIDModel({
-    ...opts,
-    fields,
-  });
-};
+export const defineLegacyVersionedModel =
+  <
+    F extends FieldDefinition,
+    IDField extends string &
+      (keyof F['generated'] | keyof F['generatedCompositeKey']),
+    SoftDeletionEnabled extends boolean,
+  >(opts: {
+    tableName: string;
+    fields: F;
+    idField: IDField;
+    softDeletionEnabled: SoftDeletionEnabled;
+  }): ModelWithIdInitializer<
+    FieldsWithVersioned<F, SoftDeletionEnabled>,
+    IDField
+  > =>
+  (masterConn, replicaConn) => {
+    const fields: ExtendedFields<F> = merge({}, opts.fields, VERSIONED_FIELDS);
+    const idModel = defineIDModel({
+      ...opts,
+      fields,
+    });
+    const model = idModel(masterConn, replicaConn);
+
+    const destroy: DestroyFn<
+      FieldsWithVersioned<F, SoftDeletionEnabled>,
+      AdditionalDestroyArgsForSequelizeTables
+    > = async (args) => {
+      const { forceHardDeletion: shouldForceHardDeletion, ...destroyArgs } =
+        args;
+      const values = {
+        deletedAt: masterConn.fn.now(3),
+        currentVersion: false,
+        latestVersion: false,
+        latestTaggedVersion: false,
+      };
+      if (
+        opts.softDeletionEnabled &&
+        !shouldForceHardDeletion &&
+        isSoftDeletionValues<FieldsWithVersioned<F, SoftDeletionEnabled>>(
+          values
+        )
+      ) {
```

Comment:
Did you test this? I think it will fail to execute because `isSoftDeletionValues()` checks that `values` is object with single key `deletedAt`, because you have this logic `Object.keys(values).length === 1 && 'deletedAt' in values`